src/client/thingtypemanager.cpp

```diff
@@ -237,6 +321,28 @@ ThingTypeList ThingTypeManager::findThingTypeByAttr(const ThingAttr attr, const
     return ret;
 }
 
+const RaceType& ThingTypeManager::getRaceData(uint32_t raceId)
+{
+    for (const auto& raceData : m_monsterRaces) {
+        if (raceData.raceId == raceId) {
+            return raceData;
+        }
+    }
+    auto emptyVal = new RaceType();
```

Comment:
Using new RaceType() introduces unnecessary complexity for managing memory, especially since RaceType is a simple structure. Instead, consider using a static empty instance, such as:

```cpp
static const RaceType emptyVal{};
return emptyVal;
```

Reply:
> is it okay now?

Yeah.